nbgrader/apps/zipcollectapp.py

```diff
@@ -0,0 +1,486 @@
+                        "submission file name."
+                        "".format(self.collector_plugin.__name__, key)
+                    )
+                if not isinstance(info[key], six.string_types):
+                    self.fail(
+                        "Expected processor {} to provide a string for {} from "
+                        "the submission file name."
+                        "".format(self.collector_plugin.__name__, key)
+                    )
+
+            root, ext = os.path.splitext(_file)
+            file_id = os.path.splitext(os.path.basename(info['file_id']))[0]
+            submission = '{}{}'.format(file_id, ext)
+            if ext in ['.ipynb'] and submission not in released_notebooks:
+                if self.strict:
+                    self.log.warn("Skipped. Invalid notebook name.")
+                    invalid_files += 1
+                    continue
+                self.log.warn("Invalid notebook name.")
+
+            if self._find_student(info) is None:
+                self.log.warn(
+                    "Skipped. Student {} not found in gradebook."
+                    "".format(info['student_id'])
+                )
+                invalid_files += 1
+                continue
+
+            submitted_path = self._format_path(
+                self.submitted_directory, info['student_id'], self.assignment_id)
+            dest_path = os.path.join(submitted_path, submission)
+
+            timestamp = self.get_timestamp()
+            if 'timestamp' in info.keys():
+                timestamp = info['timestamp'] or timestamp
+                if isinstance(timestamp, six.string_types):
+                    try:
+                        timestamp = parse_utc(timestamp)
+                    except ValueError:
+                        self.log.warn(
+                            "Could not parse the timestamp string. "
+                            "Setting timestamp to the current time."
+                        )
+
+            student_id = info['student_id']
+            if student_id in data.keys():
+                if submission not in data[student_id]['file_ids']:
+                    data[student_id]['file_ids'].append(submission)
+                    data[student_id]['timestamps'].append(timestamp)
+                    data[student_id]['src_files'].append(_file)
+                    data[student_id]['dest_files'].append(dest_path)
+                else:
+                    ind = data[student_id]['file_ids'].index(submission)
+                    old_timestamp = data[student_id]['timestamps'][ind]
+                    if timestamp >= old_timestamp:
+                        data[student_id]['timestamps'][ind] = timestamp
+                        data[student_id]['src_files'][ind] = _file
+                        data[student_id]['dest_files'][ind] = dest_path
+                    else:
+                        self.log.warn("Skipped. Older timestamp found.")
```

Comment:
In this case I think it is the warn message that is misleading. Submission files with older timestamps are skipped.

This timestamp comparison is something I would like to keep. For example if the archive file contains multiple attempts for a given submission then there will be duplicate files. I can update the code so the logic is more readable and easier to follow.